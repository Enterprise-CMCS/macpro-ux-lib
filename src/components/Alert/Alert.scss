@use "../../assets/theme/macbis-colors" as c;
// @use "../../assets/img/sprite.svg#success" as success;
// @use "../../assets/img/sprite.svg#info_alt" as info;

.usa-alert {
  button {
    border: none;
    background-color: transparent;
  }
}

.usa-alert--info:before {
  -webkit-mask: url("../../assets/img/usa-icons/info_alt.svg") no-repeat
    center/contain;
  mask: url("../../assets/img/usa-icons/info_alt.svg") no-repeat center/contain;
}

.usa-alert--success:before {
  -webkit-mask: url("../../assets/img/usa-icons/success.svg") no-repeat
    center/contain;
  mask: url("../../assets/img/usa-icons/success.svg") no-repeat center/contain;
}

.usa-alert--warning {
  background-color: c.$warn-lightest;
  border-left-color: c.$warn;
}

.usa-alert--error {
  background-color: c.$error-lightest;
  border-left-color: c.$error;
}

.usa-alert--info {
  background-color: c.$primary-alt-lightest;
  border-left-color: c.$primary-alt-dark;
}

.usa-alert--success {
  background-color: c.$success-lightest;
  border-left-color: c.$success;
}
