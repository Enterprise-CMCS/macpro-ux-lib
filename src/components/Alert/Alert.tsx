import { Icon } from "../Icon/Icon";
import React from "react";

type IntrinsicElements = JSX.IntrinsicElements["div"];

interface Props extends IntrinsicElements {
  alertHeading: string;
  alertBody: string;
  variation: "info" | "warning" | "error" | "success";
  slim?: boolean;
  icon?: boolean;
  close?: React.MouseEventHandler<HTMLButtonElement>;
}

/**
 * **Alert Component**
 *
 * @param {string}                     alertHeading   Text content for the alert heading.
 * @param {string}                     alertBody      Text content for the alert body.
 * @param {string}                     variation      Determines the alert variation to render.
 * @param {boolean}                    slim           Renders a slimmer version of the banner.
 * @param {boolean}                    icon           Show/hide the alert icon associated with the variation of the alert.
 * @param {React.MouseEventHandler}    close          Close button that shows when a event handler is passed to the component.
 */
export const Alert: React.FC<Props> = ({
  alertHeading,
  alertBody,
  variation,
  slim = false,
  icon = true,
  close,
  ...rest
}) => {
  return (
    <div
      className={`grid-row usa-alert usa-alert--${variation}${
        slim ? " usa-alert--slim" : ""
      }${icon ? "" : " usa-alert--no-icon"}`}
      {...rest}
    >
      <div className="grid-col usa-alert__body">
        <h4 className="usa-alert__heading">{alertHeading}</h4>
        <p className="usa-alert__text">{alertBody}</p>
      </div>
      {close && (
        <button
          aria-label="close alert"
          onClick={close}
          className="padding-x-2"
        >
          <Icon name="close" />
        </button>
      )}
    </div>
  );
};
