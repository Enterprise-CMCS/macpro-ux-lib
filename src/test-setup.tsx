import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import "@testing-library/jest-dom";

// Here we can set all providers for themes, stores and other stuff
const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  return <>{children}</>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };
