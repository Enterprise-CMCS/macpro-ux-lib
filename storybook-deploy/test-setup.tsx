import React, { ReactElement } from "react";
import { render, RenderOptions } from "@testing-library/react";
import "@testing-library/jest-dom";

// Here we can set all providers for themes, stores and other stuff
const AllTheProviders = ({ children }: { children: JSX.Element }) => {
  return <>{children}</>;
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, "wrapper">
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from "@testing-library/react";
export { customRender as render };

/**
 * Clean up attributes from the rendered component by removing the generated ids.
 *
 * Before:
 *   for="input-type-text-123456"
 *   id="input-type-text-123456"
 *
 * After:
 *   for="input-type-text"
 *   id="input-type-text"
 *
 * This is useful during snapshot testing. Snapshot comparisons will fail without addressing these items.
 *
 * @param {HTMLElement} container The component to clean up.
 * @param {prefixes}      string[]  The reference substrings to leave alone.
 */
