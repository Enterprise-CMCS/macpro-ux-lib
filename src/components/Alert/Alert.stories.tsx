import React from "react";
import { ComponentStory, ComponentMeta } from "@storybook/react";
import { Alert } from "./Alert";

export default {
  title: "COMPONENTS/Alert",
  component: Alert,
  args: {},
  argTypes: {},
} as ComponentMeta<typeof Alert>;

const Template: ComponentStory<typeof Alert> = ({ ...rest }) => (
  <Alert {...rest} />
);

export const Default = Template.bind({});
Default.args = {};
