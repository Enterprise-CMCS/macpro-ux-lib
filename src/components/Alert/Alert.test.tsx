import React from "react";
import { fireEvent, screen, render } from "../../test-setup";
import { Alert } from "./Alert";

describe("Tests for the Alert component", () => {
  it("should render", () => {
    render(
      <Alert
        data-testid="Alert"
        variation="info"
        alertHeading="Test"
        alertBody="Test body"
      />
    );

    const AlertComp = screen.getByTestId("Alert");

    expect(AlertComp).toBeInTheDocument();
    expect(AlertComp).toHaveTextContent("Test");
    expect(AlertComp).toHaveTextContent("Test body");
  });

  it("should render", () => {
    render(
      <Alert
        data-testid="Alert"
        variation="warning"
        alertHeading="Warning Heading"
        alertBody="Warning Body"
        icon={false}
        slim
      />
    );

    const AlertComp = screen.getByTestId("Alert");

    expect(AlertComp).toHaveTextContent("Warning Heading");
    expect(AlertComp).toHaveTextContent("Warning Body");
    expect(AlertComp).toHaveClass("usa-alert--slim");
    expect(AlertComp).toHaveClass("usa-alert--no-icon");
  });
});

describe("compontent snapshots", () => {
  it("primary big disabled", () => {
    const { container } = render(
      <Alert variation="info" alertHeading="Test1" alertBody="Test body1" />
    );
    expect(container).toMatchSnapshot();
  });

  it("secondary button", () => {
    const { container } = render(
      <Alert
        variation="warning"
        slim
        icon={false}
        alertHeading="Test2"
        alertBody="Test body2"
      />
    );
    expect(container).toMatchSnapshot();
  });
});
