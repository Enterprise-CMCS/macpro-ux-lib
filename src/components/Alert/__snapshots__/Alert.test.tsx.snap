// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compontent snapshots should render a warning alert 1`] = `
<div>
  <div
    class="grid-row usa-alert usa-alert--warning usa-alert--slim usa-alert--no-icon"
    role="alert"
  >
    <div
      class="grid-col usa-alert__body"
    >
      <h4
        class="usa-alert__heading"
      >
        Test2
      </h4>
      <p
        class="usa-alert__text"
      >
        Test body2
      </p>
    </div>
  </div>
</div>
`;

exports[`compontent snapshots should render an info alert 1`] = `
<div>
  <div
    class="grid-row usa-alert usa-alert--info"
    role="alert"
  >
    <div
      class="grid-col usa-alert__body"
    >
      <h4
        class="usa-alert__heading"
      >
        Test1
      </h4>
      <p
        class="usa-alert__text"
      >
        Test body1
      </p>
    </div>
  </div>
</div>
`;
